config/config.go

```diff
@@ -134,8 +139,45 @@ var (
 
 	// Pelican version
 	PelicanVersion string
+
+	MetadataTimeoutErr *MetadataErr = &MetadataErr{msg: "Timeout when querying metadata"}
 )
 
+// This function creates a new MetadataError by wrapping the previous error
+func NewMetadataError(err error, msg string) *MetadataErr {
+	return &MetadataErr{
+		msg: msg,
+	}
+}
+
+func (e *MetadataErr) Error() string {
+	// If the inner error is nil, we don't want to print out "<nil>"
+	if e.innerErr != nil {
+		return fmt.Sprintf("%s: %v", e.msg, e.innerErr)
+	} else {
+		return e.msg
+	}
+}
+
+func (e *MetadataErr) Is(target error) bool {
+	// We want to verify we have a timeout error
+	if _, ok := target.(*MetadataErr); ok {
+		return e.msg == target.Error()
```

Comment:
Ah yeah, you're right. In the earlier comment I was not sure because `target` is of type `error` but we literally convert it right there lol. Fixed now!